fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -0,0 +1,274 @@
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
+        output.append("\n");
+
+        return output.toString();
+    }
+
+    @Override
+    protected String getNodeVerboseExplain(String prefix) {
+        StringBuilder output = new StringBuilder();
+
+        output.append(prefix).append("TABLE: ").append(srIcebergTable.getName()).append("\n");
+
+        if (null != sortColumn) {
+            output.append(prefix).append("SORT COLUMN: ").append(sortColumn).append("\n");
+        }
+        if (!conjuncts.isEmpty()) {
+            output.append(prefix).append("PREDICATES: ").append(
+                    getExplainString(conjuncts)).append("\n");
+        }
+
+        output.append(prefix).append(String.format("avgRowSize=%s", avgRowSize));
+        output.append("\n");
+
+        output.append(prefix).append(String.format("numNodes=%s", numNodes));
+        output.append("\n");
+
+        return output.toString();
+    }
+
+    @Override
+    public int getNumInstances() {
+        return result.size();
+    }
+
+    @Override
+    public void computeStats(Analyzer analyzer) {
+        Optional<Snapshot> snapshot = IcebergUtil.getCurrentTableSnapshot(
+                srIcebergTable.getIcebergTable(), true);
+        if (snapshot.isPresent()) {
+            for (FileScanTask task : IcebergUtil.getTableScan(
+                    srIcebergTable.getIcebergTable(), snapshot.get(),
+                    icebergPredicates, true).planFiles()) {
+                totalBytes += task.length();
+                DataFile file = task.file();
+                cardinality += file.recordCount() * (task.length() / file.fileSizeInBytes());
+                LOG.info("Total bytes " + totalBytes + ", Cardinality " + cardinality);
+            }
+        }
+        if (cardinality > 0) {
+            avgRowSize = totalBytes / (float) cardinality;
+            if (hasLimit()) {
+                cardinality = Math.min(cardinality, limit);
+            }
+
+            numNodes = Math.min(hostToBeId.size(), result.size());
+        }
+        // even current node scan has no data, at least one backend will be assigned when the fragment actually execute
+        numNodes = numNodes <= 0 ? 1 : numNodes;
+        // when node scan has no data, cardinality should be 0 instead of a invalid value after computeStats()
+        cardinality = cardinality == -1 ? 0 : cardinality;
+    }
+
+    @Override
+    protected void toThrift(TPlanNode msg) {
+        msg.node_type = TPlanNodeType.HDFS_SCAN_NODE;
+        THdfsScanNode tHdfsScanNode = new THdfsScanNode();
+        tHdfsScanNode.setTuple_id(desc.getId().asInt());
+        msg.hdfs_scan_node = tHdfsScanNode;
+
+        if (msg.isSetConjuncts()) {
+            msg.conjuncts.clear();
+        }
+        for (Expr expr : conjuncts) {
+            msg.addToConjuncts(expr.treeToThrift());
+        }
```

Comment:
`conjuncts` has been already added into `TPlanNode` in base class `PlanNode`, so this can be removed.